src/lvl_script_commands.c

```diff
@@ -5162,6 +5174,87 @@ static void set_increase_on_experience_process(struct ScriptContext* context)
     }
 }
 
+static void set_player_modifier_check(const struct ScriptLine* scline)
+{
+    ALLOCATE_SCRIPT_VALUE(scline->command, 0);
+    PlayerNumber plyr_idx = scline->np[0];
+    short mdfrdesc = get_id(modifier_desc, scline->tp[1]);
+    short mdfrval = scline->np[2];
+    const char *mdfrname = modifier_desc[mdfrdesc - 1].name;
+    if (mdfrdesc == -1)
+    {
+        SCRPTERRLOG("Unknown Player Modifier '%s'.", mdfrname);
+        DEALLOCATE_SCRIPT_VALUE
+        return;
+    }
+    if (mdfrval < 0)
+    {
+        SCRPTERRLOG("Value %d out of range for Player Modifier '%s'.", mdfrval, mdfrname);
+        DEALLOCATE_SCRIPT_VALUE
+        return;
+    }
+    struct Dungeon* dungeon = get_dungeon(plyr_idx);
+    if (dungeon_invalid(dungeon))
+    {
+        SCRPTERRLOG("Can't manipulate Player Modifier '%s', player %d has no dungeon.", mdfrname, (int)plyr_idx);
+        DEALLOCATE_SCRIPT_VALUE
+        return;
+    }
+    value->shorts[0] = plyr_idx;
+    value->shorts[1] = mdfrdesc;
+    value->shorts[2] = mdfrval;
+    PROCESS_SCRIPT_VALUE(scline->command);
+}
+
+static void set_player_modifier_process(struct ScriptContext* context)
+{
+    PlayerNumber plyr_idx = context->value->shorts[0];
```

Comment:
I believe other functions use a `for (int plyr_idx = context->plr_start; plyr_idx < context->plr_end; plyr_idx++)` loop for players instead of context->value->shorts[0];
I suppose atm it won't work with ALL_PLAYERS, while there's no practical reason it shouldn't